File: cis_interface/examples/SaM/src/SaM.py
#!/usr/bin/python
from yggdrasil.interface.YggInterface import YggInput, YggOutput


if __name__ == '__main__':

    # Get input and output channels matching yaml
    in1 = YggInput('input1_python')
    in2 = YggInput('static_python')
    out1 = YggOutput('output_python')
    print('SaM(P): Set up I/O channels')

    # Get input from input1 channel
    ret, adata = in1.recv()
    if not ret:
        raise RuntimeError('SaM(P): ERROR RECV from input1')
    a = int(adata)
    print('SaM(P): Received %d from input1' % a)

    # Get input from static channel
    ret, bdata = in2.recv()
    if not ret:
        raise RuntimeError('SaM(P): ERROR RECV from static')
    b = int(bdata)
    print('SaM(P): Received %d from static' % b)

    # Compute sum and send message to output channel
    sum = a + b
    outdata = '%d' % sum
    ret = out1.send(outdata)
    if not ret:
        raise RuntimeError('SaM(P): ERROR SEND to output')
    print('SaM(P): Sent to output')

File: cis_interface/examples/SaM/src/SaM.m

% Get input and output channels matching yaml
in1 = YggInterface('YggInput', 'input1_matlab');
in2 = YggInterface('YggInput', 'static_matlab');
out1 = YggInterface('YggOutput', 'output_matlab');
disp('SaM(M): Set up I/O channels');

% Get input from input1 channel
[flag, var] = in1.recv();
if (~flag);
  error('SaM(M): ERROR RECV from input1');
end
a = str2num(var);
fprintf('SaM(M): Received %d from input1\n', a);

% Get input from static channel
[flag, var] = in2.recv();
if (~flag);
  error('SaM(M): ERROR RECV from static');
end
b = str2num(var);
fprintf('SaM(M): Received %d from static\n', b);

% Compute sum and send message to output channel
sum = a + b;
ret = out1.send(int2str(sum));
if (~ret);
  error('SaM(M): ERROR SEND to output');
end
disp('SaM(M): Sent to output');
